core/shared/src/main/scala/com/lynbrookrobotics/potassium/PeriodicSignal.scala

```diff
@@ -125,6 +125,29 @@ abstract class PeriodicSignal[+T] { self =>
     }
   }
 
+  /**
+    * Applies a timestamped fixed size sliding window over the signal
+    * @param size the size of the window
+    * @param filler the element to use to fill the window until elements are available
+    * @return
+    */
+  def slidingWithTime[U >: T](size: Int, filler: U): PeriodicSignal[Queue[(Time, U)]] = {
+    // TODO: poor resolution with currentTimeMillis
```

Comment:
Agree, maybe we can introduce a way for the clock to provide the system time? And so this could take an implicit clock to read time from.